src/mixxx.cpp

```diff
@@ -1154,36 +1154,42 @@ void MixxxMainWindow::slotOptionsPreferences() {
 }
 
 void MixxxMainWindow::slotNoVinylControlInputConfigured() {
-    QMessageBox::warning(
+    QMessageBox::StandardButton btn = QMessageBox::warning(
         this,
         Version::applicationName(),
         tr("There is no input device selected for this vinyl control.\n"
-           "Please select an input device in the sound hardware preferences first."),
-        QMessageBox::Ok, QMessageBox::Ok);
-    m_pPrefDlg->show();
-    m_pPrefDlg->showSoundHardwarePage();
+           "Configure an input device in the sound hardware preferences now?"),
+        QMessageBox::Yes | QMessageBox::Cancel, QMessageBox::Cancel);
+    if (btn == QMessageBox::Yes) {
```

Comment:
this works as expectd on  Linux, though I'm not familiar with Qt messageboxes.. Is this correct?

Reply:
Yes, thank you.